capstone/src/main/java/com/google/sps/servlets/student/StudentServlet.java

```diff
@@ -91,16 +88,31 @@ private static String convertToJsonUsingGson(StudentInfo info) {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // TODO: All commented steps below
-    // 1. Add club to logged in student's club list if requested
-    // 2. Remove club from logged in student's club list if requested
-    // 3. Update student information with edited content
+    // 1. Remove club from logged in student's club list if requested
+    // 2. Update student information with edited content
 
     // No need to check if user's information is not in Datastore - this is done in doGet method
     // Get student object based on the logged in email
     UserService userService = UserServiceFactory.getUserService();
     String userEmail = userService.getCurrentUser().getEmail();
 
-    response.sendRedirect("/profile.html");
+    Query query = new Query(userEmail);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    Entity student = ImmutableList.copyOf(results.asIterable()).get(0);
+
+    String clubToJoin = request.getParameter(Constants.JOIN_CLUB_PROP);
+    if (clubToJoin != null && !clubToJoin.isEmpty()) {
+      // Add member to club's member list and update Datastore
+      Entity club = retrieveClub(clubToJoin, datastore);
+      addItemToEntity(club, datastore, userEmail, Constants.MEMBER_PROP);
+
+      // Add new club to student's club list and update Datastore
+      addItemToEntity(student, datastore, clubToJoin, Constants.PROPERTY_CLUBS);
+      response.sendRedirect("index.html");
```

Comment:
I remember we talked abt having some sort of confirmation when the user joined the club. do u think redirecting is enough/is there other info being communicated to the user?

Reply:
I think we talked about changing the "Join Club" to "Leave Club" so I can do that once the "Leave Club" button is fully functioning. I wasn't exactly sure if we could redirect to the club's About Us page since the URLSearchParams is in JS, do you know if there's a way to do this?

Reply:
Yeah, actually I do this in my servlet that I'm currently working on. Instead of just redirecting to about-us.html, you can use a query search param like this:
` response.sendRedirect("/about-us.html?name=" + clubEntity.getProperty(Constants.PROPERTY_NAME)); `
up to you if you want to actually redirect to about-us.html though.

Reply:
Awesome, just changed it so "Join Club" also redirects to the club's page. Thanks Megan!